relay.go

```diff
@@ -582,13 +588,13 @@ func (r *Relayer) handleNonCallReq(f *Frame) error {
 	sent, failure := item.destination.Receive(f, frameType)
 	if !sent {
 		r.failRelayItem(items, originalID, failure, errFrameNotSent)
-		return nil
+		return _relayShouldRelease, nil
 	}
 
 	if finished {
 		r.finishRelayItem(items, originalID)
 	}
-	return nil
+	return !sent, nil
```

Comment:
similar to above, `!sent` is already handled, so to get here, `!sent` must be false, so simplify

Reply:
updated. thanks!